### PR TITLE
add username password auth

### PR DESCRIPTION
**Pull request by gangtao**
1. add username password based authenication 
2. update test, no external dependecy other than kafka

there are two updates for generated code
1. https://github.com/timeplus-io/gluon/pull/94/files#diff-7950f9235f189264122b2be7c57b0d6d4ef31f3f6a1737e5302f4b6bcbe3d4caR79 skip the close warning
2. https://github.com/timeplus-io/gluon/pull/94/files#diff-7950f9235f189264122b2be7c57b0d6d4ef31f3f6a1737e5302f4b6bcbe3d4caR141 add username/password authentication

the second change is not nessesary in case our swagger doc has the basic authentication specified @ye11ow

### CODE REVIEW

File: python/timeplus/dbapi.py [hunk 1]
```diff
@@ -84,16 +85,24 @@ def connect(
     Connection: A Connection object.
     """
     address = f"{scheme}://{host}:{port}"
-    apikey = password
     workspace = path
-    return Connection(address, apikey, workspace)
+    if user is not None and password is not None:
```
Comment (ye11ow) [inline python/timeplus/dbapi.py:89]: Proton doesn't require password so, password can be `None` here.

File: python/timeplus/dbapi.py [hunk 2]
```diff
@@ -102,8 +111,24 @@ def __init__(
...
+
+        self.env = Environment().address(address).workspace(workspace)
+
+        if apikey is not None:
+            self.env.apikey(apikey)
+
+        if username is not None:
+            self.env.username(username)
+
+        if password is not None:
```
Comment (ye11ow) [inline python/timeplus/dbapi.py:129]: it can be none

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -52,6 +52,8 @@ jobs:
           TIMEPLUS_API_KEY: ${{ secrets.TIMEPLUS_API_KEY }}
           TIMEPLUS_HOST: ${{ secrets.TIMEPLUS_HOST }}
           TIMEPLUS_WORKSPACE: ${{ secrets.TIMEPLUS_WORKSPACE }}
+          EXTERNAL_STREAM_PASSWORD: ${{ secrets.EXTERNAL_STREAM_PASSWORD }}
+          TIMEPLUS_REPLICATION_NUMBER: 3
```
Comment (ye11ow) [inline .github/workflows/python-package.yml:56]: `TIMEPLUSD_REPLICATION_NUMBER` ?

File: python/timeplus/dbapi.py (hunk 1 above)
Comment (gangtao) [reply python/timeplus/dbapi.py:89]: is that empty not None?

File: .github/workflows/python-package.yml (hunk 3 above)
Comment (gangtao) [reply .github/workflows/python-package.yml:56]: I think it is OK, as the other component wont do replication

File: python/timeplus/dbapi.py (hunk 1 above)
Comment (gangtao) [reply python/timeplus/dbapi.py:89]: I assum in case the password in None, it equals no authentication?

File: python/timeplus/dbapi.py (hunk 2 above)
Comment (gangtao) [reply python/timeplus/dbapi.py:129]: if the password is None, dont set it is OK?

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**